### PR TITLE
Small refactor in highlight

### PR DESCRIPTION
**Pull request by kaste**
Remove some cruft from `highlight.py`

- The `reset` methods were unused
- The `clear` methods were duplicated
- `persist.highlights` is unused
- The RegionStore is a singleton, only used within *highlight.py* itself, so I removed it from *persist* as well.

### CODE REVIEW

File: lint/highlight.py [hunk 1]
```diff
@@ -22,12 +22,19 @@
...
+    @property
+    def memory(self):
         """structure: {"view.id": [region_keys ... ]}."""
-        self.memory = sublime.load_settings("sl_regions.sublime-settings")
-        views = self.memory.get("views")
-        if not views:
-            self.memory.set("views", {})
+        memory = self._memory
+        if not memory:
+            memory = sublime.load_settings("sl_regions.sublime-settings")
```
Comment (kaste) [inline lint/highlight.py]: Don'

Comment (kaste) [reply lint/highlight.py]: We never call `save_settings` if that matters.

Comment (braver) [reply lint/highlight.py]: first question: I don't know

second: we only ever *read* settings, so we shouldn't have a need for `save_settings `

Comment (kaste) [reply lint/highlight.py]: But it looks like someone wanted to persist these regions across sessions. And we could use `save_settings` to do that. But it is not implemented. @pykong did this.

If we just use a normal *dict*, we don't need a lazy *property* here.

Comment (braver) [reply lint/highlight.py]: Right, that sounds like a remnant of our earlier iterations. I think SL3 had a lot of caching and persistence, etc., that we don't need anymore. IMO you're free to change this.

Comment (pykong) [reply lint/highlight.py]: @kaste @braver we could change that indeed if everything works as before.

The idea behind the in-memory settings was to persist region throughout a session.
So, when there is an exception that SublimeLinter will be given the chance to pick up those then remnant highlights from the previous run and clear them.

**Comment by braver** [comment]
@kaste is this still WIP?

**Comment by kaste** [comment]
Done. 

I found a bug here as well. https://github.com/SublimeLinter/SublimeLinter/pull/766/commits/23e431c48ae526181043b245a11b593841f4192f

**Pull request merged**